plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.js

```diff
@@ -0,0 +1,639 @@
+    }
+
+    this.dropdownContainer_ = container;
+
+    this.updateSliderValues_();
+  }
+
+  /**
+   * Disposes of events and DOM-references belonging to the colour editor.
+   * @private
+   */
+  dropdownDisposeSliders_() {
+    for (const event of this.boundEvents_) {
+      Blockly.browserEvents.unbind(event);
+    }
+    this.boundEvents_.length = 0;
+    this.hueReadout_ = null;
+    this.hueSlider_ = null;
+    this.saturationReadout_ = null;
+    this.saturationSlider_ = null;
+    this.brightnessReadout_ = null;
+    this.brightnessSlider_ = null;
+    this.dropdownContainer_ = null;
+  }
+
+  /**
+   * Updates the value of this field based on the editor sliders.
+   * @param {?Event} event Unused.
+   * @private
+   */
+  onSliderChange_(event) {
+    const hue = parseFloat(this.hueSlider_.value);
+    const saturation = parseFloat(this.saturationSlider_.value);
+    const brightness = parseFloat(this.brightnessSlider_.value);
+    const hsv = new HsvColour(
+        hue / 360.0, saturation / 100.0, brightness / 100.0);
+    const rgb = new RgbColour().loadFromHsv(hsv);
+    this.setValue(rgb.toHex());
+    this.renderSliders_();
+  }
+
+  /**
+   * Updates the value of this field and editor sliders using an eyedropper.
+   * @param {?Event} event Unused.
+   * @private
+   */
+  onEyedropperEvent_(event) {
+    const eyeDropper = new window.EyeDropper();
+    eyeDropper.open().then((result) => {
+      this.setValue(result.sRGBHex);
+      this.updateSliderValues_();
+    });
+  }
+
+  /**
+   * Updates the value of this field and editor sliders based on a colour input.
+   * @param {?InputEvent} event The event generated from editing an input
+   *   element.
+   * @private
+   */
+  onBrowserColourInputEvent_(event) {
+    this.setValue(event.target.value);
+    this.updateSliderValues_();
+  }
+
+  /**
+   * Updates the gradient backgrounds of the slider tracks and readouts based
+   * on the slider values.
+   * @private
+   */
+  renderSliders_() {
+    const hue = parseFloat(this.hueSlider_.value);
+    const saturation = parseFloat(this.saturationSlider_.value);
+    const brightness = parseFloat(this.brightnessSlider_.value);
+
+    this.hueReadout_.textContent = String(hue);
+    this.saturationReadout_.textContent = String(saturation);
+    this.brightnessReadout_.textContent = String(brightness);
+
+    const hsv = new HsvColour(
+        hue / 360.0, saturation / 100.0, brightness / 100.0);
+    // Temporary colour structures to allow manipulation in the HSV colour space
+    // and conversion to the RGB colour space.
+    const tempHsv = new HsvColour();
+    const tempRgb = new RgbColour();
+    const trackRadius = '8px';
+
+    let hueGradient = 'linear-gradient(to right, ';
+    tempHsv.copy(hsv).setHue(0.0/6.0);
+    hueGradient += tempRgb.loadFromHsv(tempHsv).toHex() +
+        ' ' + trackRadius + ', ';
+    tempHsv.copy(hsv).setHue(1.0/6.0);
+    hueGradient += tempRgb.loadFromHsv(tempHsv).toHex() + ', ';
```

Comment:
I think the stateful string building is a bit difficult to parse. Maybe you could separate this into a helper function and modify the below code to use template strings? Eg `hsvToHex(hue, sat, vib): string`

Reply:
Done, I added a private static helper method hsvToHex_.